src/App.tsx

```diff
@@ -1,26 +1,227 @@
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+            data-cy="ToggleAllButton"
+          />
+
+          <form onSubmit={handleSubmit}>
+            <input
+              value={inputValue}
+              onChange={event => setInputValue(event.target.value)}
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              ref={inputRef}
+              disabled={!!tempTodo}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main" data-cy="TodoList">
+          {filteredTodos.map(todo => (
+            <CompletedTodo
+              key={todo.id}
+              todo={todo}
+              onDelete={onDeleteTodo}
+              isLoading={loadingTodoId.includes(todo.id)}
+            />
+          ))}
+          {tempTodo && (
+            <CompletedTodo todo={tempTodo} isLoading onDelete={onDeleteTodo} />
+          )}
+        </section>
 
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {unCompletedTodosCounter} items left
+            </span>
+            <nav className="filter" data-cy="Filter">
+              <button
+                className={classNames('filter__link', {
+                  selected: filter === 'all',
+                })}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilter(Filter.All)}
+              >
+                All
+              </button>
+
+              <button
+                className={classNames('filter__link', {
+                  selected: filter === 'active',
+                })}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilter(Filter.Active)}
+              >
+                Active
+              </button>
+
+              <button
+                className={classNames('filter__link', {
+                  selected: filter === 'completed',
+                })}
+                data-cy="FilterLinkCompleted"
+                onClick={() => setFilter(Filter.Completed)}
+              >
+                Completed
```

Comment:
Mentors will tell you to map this buttons)

Reply:
Use `map` 😄